pkg/controller/daemon/update.go

```diff
@@ -372,7 +373,8 @@ func (dsc *DaemonSetsController) dedupCurHistories(ds *apps.DaemonSet, curHistor
 					toUpdate.Labels = make(map[string]string)
 				}
 				toUpdate.Labels[apps.DefaultDaemonSetUniqueLabelKey] = keepCur.Labels[apps.DefaultDaemonSetUniqueLabelKey]
-				_, err = dsc.kubeClient.CoreV1().Pods(ds.Namespace).Update(context.TODO(), toUpdate, metav1.UpdateOptions{})
+				patchJson := fmt.Sprintf(`{"metadata": {"labels": {"%s": "%s"}}}`, apps.DefaultDaemonSetUniqueLabelKey, keepCur.Labels[apps.DefaultDaemonSetUniqueLabelKey])
+				_, err = dsc.kubeClient.CoreV1().Pods(ds.Namespace).Patch(context.TODO(), toUpdate.Name, types.StrategicMergePatchType, []byte(patchJson), metav1.PatchOptions{})
```

Comment:
Use pod Name instead of `toUpdate.Name` here. Then, lines 371-375 can be removed.